controller/lib/hal/hal_stm32.cpp

```diff
@@ -0,0 +1,856 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#if defined(BARE_STM32)
+
+#include "hal.h"
+#include "stm32.h"
+
+// This is the main stack 
+uint32_t mainStack[ 500 ];
```

Comment:
I personally don't mind this formatting, but we have agreed on clang-format as the source of truth and we have precommits enforcing it - e.g. if someone changes this code, their diff will be contaminated by additional autoformatting changes if we don't format this file now. Please follow the instructions for installing precommit hooks and reformat this file.

https://github.com/RespiraWorks/VentilatorSoftware/wiki#precommit-hooks